pg_shard.c

```diff
@@ -586,6 +607,25 @@ SelectFromMultipleShards(Query *query, List *queryShardList)
 
 
 /*
+ * TODO: Add documentation
+ */
+static Query *
+BuildLocalQuery(Query *query, List *localRestrictList)
+{
+	Query *localQuery = copyObject(query);
+	FromExpr *joinTree = localQuery->jointree;
+
+	if (joinTree != NULL)
```

Comment:
Should this be an assert? If the joinTree is NULL, we aren't updating the clauses, but returning the query as-is. We should either: a) Error out if we expect joinTree to be NULL, b) Assert if we expect it to be there, c) Create a joinTree if its NULL and set the clauses?